src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,26 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch(Exception ex)
```

Comment:
I think "because I'm not brave enough" is my best "why not" right now.  It would be a pity if `SerialPort` stayed forever as a blob of early 21st century flakiness, but I think at the moment the best bang-for-buck comes from simple robust palliatives rather than proper rework.  To have *any* improvement to the problems people have been screaming about for a decade is going to be a bit of a revelation.  #16261 is a difficult async close/reopen bug which may well benefit from better cleanup of this background thread, so I think moving to Task.Run will be a good idea eventually.  How to avoid getting jammed-up by blocking in SerialPort.Dispose will be a challenge at that point.